_delphi_utils_python/delphi_utils/flagging/flag_io.py

```diff
@@ -0,0 +1,369 @@
+"""Functions related to determining which files to generate with which parameters."""
+import os
+import glob
+from datetime import timedelta, date
+import numpy as np
+import pandas as pd
+import boto3
+import covidcast
+from .generate_reference import gen_ref_dfs
+from .generate_ar import gen_ar_files
+from .. import (
+    get_structured_logger,
+)
+
+
+def gen_files(gen_ref_dict:dict, gen_ar_dict:dict, logger) -> dict:
+    """
+    Generate necessary files.
+
+    This method takes in dictionaries that correspond to the files needed to
+    be generated that are either reference files or the autoregression files.
+
+    The output is a complete dictionary of all the file names and their respective dataframes.
+    """
+    files_dict = {}
+    for key, params in gen_ref_dict.items():
+        df_dict = gen_ref_dfs(params['raw_df'].query("lag==@params['lag']")
+                              .drop(columns=['lag']), logger)
+        files_dict.update(df_dict)
+        files_dict.update(gen_ar_files(key,
+                   df_dict['ref_dfs/wkdy_corr.csv'],
+                   params['ar_lag'], params['n_train'],
+                   params['resid_dist_dates'], params['eval_dates']))
+    for key, params in gen_ar_dict.items():
+        ar_dict = gen_ar_files(key, params['input_df'],
+                                       params['ar_lag'], params['n_train'],
+                                       params['resid_dist_dates'], params['eval_dates'])
+        files_dict.update(ar_dict)
+    return files_dict
+
+
+def missing_local_files(df: pd.DataFrame, flag_p: dict, flag_meta: dict) -> (dict, dict):
+    """
+    Find files that are missing when this method is running on a local filesystem.
+
+    Note that there are different methods for local and remote because the s3 filestructure is flat.
```

Comment:
I think the issue is because python's os module, which I use for local file handling, wouldn't be able to traverse the s3 filesystem because of the flat nature. But it would be great only to have one method - I'll think about alternatives.